### PR TITLE
NIFI-5753 Add SSL support to HortonworksSchemaRegistry service

### PR DESCRIPTION
**Pull request by grzegorz8**
Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [x] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [x] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [x] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by joewitt** [comment]
thanks for the contrib.  i've not tested it on a live/secured instance.  Have you done so?

**Comment by grzegorz8** [comment]
@joewitt It's already running on my DEV nifi cluster. No problems so far.

File: nifi-nar-bundles/nifi-standard-services/nifi-hwx-schema-registry-bundle/nifi-hwx-schema-registry-service/src/main/java/org/apache/nifi/schemaregistry/hortonworks/HortonworksSchemaRegistry.java [hunk 1]
```diff
@@ -120,9 +131,31 @@ public void enable(final ConfigurationContext context) throws InitializationExce
...
     }
 
-
+    private Map<String, String> buildSslProperties(final ConfigurationContext context) {
+        final SSLContextService sslContextService = context.getProperty(SSL_CONTEXT_SERVICE).asControllerService(SSLContextService.class);
+        ImmutableMap.Builder<String, String> propertiesBuilder = ImmutableMap.builder();
+        if (sslContextService != null) {
+            propertiesBuilder.put("protocol", sslContextService.getSslAlgorithm());
+            propertiesBuilder.put("keyPassword", sslContextService.getKeyPassword());
+            if (sslContextService.isKeyStoreConfigured()) {
```
Comment (MikeThomsen) [inline nifi-nar-bundles/nifi-standard-services/nifi-hwx-schema-registry-bundle/nifi-hwx-schema-registry-service/src/main/java/org/apache/nifi/schemaregistry/hortonworks/HortonworksSchemaRegistry.java]: The registry client appears to be pretty temperamental about you not supplying both a keystore and a trust store. So I think you're going to need to add a `customValidate` validation step to verify that the SSL context is set up the way that the HWX registry is expecting. You can search the code and find examples, but they look like this on the method signature:

```
protected Collection<ValidationResult> customValidate(ValidationContext context) {

}
```

**Comment by bbende** [comment]
Tested this PR and it appears to work with the minor issues mentioned by Mike above.

I'm going to add an additional commit with the following two minor changes...

1) Move the use of keyPassword inside of the if block for when a keystore is provided because you can't have a keyPassword unless you also have a keystore, and also make keypassword optional, so it looks like this:
```
if (sslContextService.isKeyStoreConfigured()) {
        propertiesBuilder.put("keyStorePath", sslContextService.getKeyStoreFile());
        propertiesBuilder.put("keyStorePassword", sslContextService.getKeyStorePassword());
        propertiesBuilder.put("keyStoreType", sslContextService.getKeyStoreType());
        if (sslContextService.getKeyPassword() != null) {
            propertiesBuilder.put("keyPassword", sslContextService.getKeyPassword());
        }
}
```

2. Bump the client version from 0.5.3 to 0.8.0 (latest)

Regarding Mike's other comment about customValidate, this is really incorrect behavior of the schema registry client to require a keystore. I don't think NiFi should implement the logic to force users into that since ideally the schema registry client would get updated to make keystore optional. I will report that issue to the schema registry projected.

**Pull request closed without merging**